docs/advanced/customization-guide.md

```diff
@@ -16,8 +15,1007 @@ published: false
+trying to execute them. A subdirectory under the hooks directory can be used,
+for example `/etc/kubernetes/node-feature-discovery/source.d/conf/`.
+
+**NOTE:** NFD will blindly run any executables placed/mounted in the hooks
+directory. It is the user's responsibility to review the hooks for e.g.
+possible security implications.
+
+**NOTE:** The [minimal](deployment-and-usage#minimal) image variant only
+supports running statically linked binaries.
+
+### Feature files
+
+The *local* source reads files found in
+`/etc/kubernetes/node-feature-discovery/features.d/`.
+
+### Input format
+
+The hook stdout and feature files are expected to contain features in simple
+key-value pairs, separated by newlines:
+
+```plaintext
+<name>[=<value>]
+```
+
+The label value defaults to `true`, if not specified.
+
+Label namespace may be specified with `<namespace>/<name>[=<value>]`. The
+namespace must be explicitly allowed with the `-extra-label-ns` command line
+flag of nfd-master if using something else than
+`[<sub-ns>.]feature.node.kubernetes.io` or
+`[<sub-ns>.]profile.node.kubernetes.io`.
+
+### Mounts
+
+The standard NFD deployments contain `hostPath` mounts for
+`/etc/kubernetes/node-feature-discovery/source.d/` and
+`/etc/kubernetes/node-feature-discovery/features.d/`, making these directories
+from the host available inside the nfd-worker container.
+
+#### Injecting labels from other pods
+
+One use case for the hooks and/or feature files is detecting features in other
+Pods outside NFD, e.g. in Kubernetes device plugins. By using the same
+`hostPath` mounts for `/etc/kubernetes/node-feature-discovery/source.d/` and
+`/etc/kubernetes/node-feature-discovery/features.d/` in the side-car (e.g.
+device plugin) creates a shared area for deploying hooks and feature files to
+NFD. NFD will periodically scan the directories and run any hooks and read any
+feature files it finds.
+
+## Custom feature source
+
+The *custom* feature source in nfd-worker provides a rule-based mechanism for
+label creation, similar to the
+[NodeFeatureRule](#nodefeaturerule-custom-resource) objects. The difference is
+that the rules are specified in the worker configuration instead of a
+Kubernetes API object.
+
+See [worker configuration](deployment-and-usage.md#worker-configuration) for
+instructions how to set-up and manage the worker configuration.
+
+### An example custom feature source configuration
+
+Consider the following referential configuration for nfd-worker:
+
+```yaml
+core:
+  labelSources: ["custom"]
+sources:
+  custom:
+    - name: "my sgx feature rule"
+      labels:
+        "my-sgx-feature": "true"
+      matchFeatures:
+        - feature: cpu.cpuid
+          matchExpressions:
+            SGX: {op: Exists}
+        - feature: kernel.config
+          matchExpressions:
+            X86_SGX: {op: In, value: ["y"]}
+```
+
+It specifies one rule which creates node label
+`feature.node.kubenernetes.io/my-sgx-feature=true` if both of the following
```

Comment:
`custom` rule adds it's own prefix:
`feature.node.kubenernetes.io/custom-my-sgx-feature=true`

Reply:
It shouldn't anymore, after #697.

The legacy `matchOn` rules add the prefix, though. Maybe we should add a note about this 🤔

Reply:
Good to know!